### PR TITLE
Refactor assignments settings to accept json

### PR DESCRIPTION
**Pull request by ksimuk**
* convert setings structure
* add get method

### CODE REVIEW

File: src/lib/assignment.ts [hunk 1]
```diff
@@ -145,14 +181,103 @@ async function reducePublish(courseId: string, srcDir: string, yamlDir: string,
...
+      disableDownloadByStudent: res.disableDownloadByStudent,
+      visibilityOnDisabled: res.visibilityOnDisabled,
+      visibilityOnCompleted: res.visibilityOnCompleted,
+      startTime: convertDateToLocal(res.startTime),
+      endTime: convertDateToLocal(res.endTime),
+      action: res.action,
+      penalties: res.penalties? _.map(res.penalties, _ => {
+        return {
+          id: _.id,
+          datetime: new Date(_.datetime),
```
Comment (AndreyNikitin) [inline src/lib/assignment.ts:256]: Different approach new Date() or _.toIsoString

Comment (ksimuk) [reply src/lib/assignment.ts:256]: AssignmentSettings uses `Date`, API returns and works with `string`

**Review by AndreyNikitin — APPROVED** [review]

File: src/lib/assignment.ts [hunk 2]
```diff
@@ -21,6 +21,43 @@ type Yaml = {
...
+  id: number
+  datetime: string
+  percent: number
+  message: string
+}
+
+export type AssignmentSettings = {
+  enableResetAssignmentByStudent?: boolean
+  disableDownloadByStudent?: boolean
+  visibilityOnDisabled?: string, // "READ_ONLY", "NO_ACCESS",
```
Comment (destitutus) [inline src/lib/assignment.ts:41]: probably could be string literal type (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-1-8.html#string-literal-types), like
`visibilityOnDisabled?: 'READ_ONLY' | 'NO_ACCESS'
the same for other includes

Comment (ksimuk) [reply src/lib/assignment.ts:41]: was thinking about it, but didn't want to introduce a new version of the library with a string parameter value change.

**Review by AndreyNikitin — APPROVED** [review]

**Pull request merged**